### PR TITLE
backuppcl: pass finishesSpec field to resume span

### PR DESCRIPTION
**Pull request by msbutler**
PR #114268 broke the plumbing for the CompletedSpans metric which allows the backup coordinator to update the FractionCompleted metric. This patch fixes this bug by passing the finishesSpec field to the resume span.

Informs #120161

Release note: none

### CODE REVIEW

File: pkg/ccl/backupccl/backup_processor.go [hunk 1]
```diff
@@ -636,12 +636,13 @@ func runBackupProcessor(
...
-								start:      span.start,
-								end:        span.end,
-								attempts:   span.attempts,
-								lastTried:  span.lastTried,
+								span:         *resp.ResumeSpan,
+								firstKeyTS:   resumeTS,
+								start:        span.start,
+								end:          span.end,
+								attempts:     span.attempts,
+								lastTried:    span.lastTried,
```
Comment (stevendanna) [inline pkg/ccl/backupccl/backup_processor.go:644]: Should we reconsider our pattern her. Perhaps we should copy the whole struct and then reset the fields we want to reset.

Comment (msbutler) [reply pkg/ccl/backupccl/backup_processor.go:644]: perhaps in a follow up pr that doesn't get backported?

Comment (msbutler) [reply pkg/ccl/backupccl/backup_processor.go]: done.

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=dt

**Comment by dt** [comment]
When you manually backport to 23.1, can you do 23.1.17-rc as well?

**Pull request merged**